4_Largest_Palindrome_Product.py

```diff
@@ -0,0 +1,27 @@
+
+num1 = 999
+max_palindrome = 0
+while num1 > 0:
+    num2 = 999
+    while num2 > 0:
+
+        product = num1 * num2
+        i = 0
+        end = -1
+        palindrome = True
+        while i < len(str(product))/2:
+            if str(product)[i] != str(product)[end]:
```

Comment:
I don't know how much optimization the Python interpreter does, but here the `str(product)` computation might be done repeatedly when you really only need to do it once up front.